SU2_CFD/src/python_wrapper_structure.cpp

```diff
@@ -31,865 +31,644 @@
+    
+    iPoint = geometry_container[ZONE_0][INST_0][MESH_0]->vertex[iMarker][iVertex]->GetNode();
+    
+    if(geometry_container[ZONE_0][INST_0][MESH_0]->nodes->GetDomain(iPoint) && compressible){
+        vertexWallTemp = solver_container[ZONE_0][INST_0][MESH_0][FLOW_SOL]->GetNodes()->GetTemperature(iPoint);
+    }
+    
+    return SU2_TYPE::GetValue(vertexWallTemp);
+    
 }
 
 void CDriver::SetVertexTemperature(unsigned short iMarker, unsigned long iVertex, passivedouble val_WallTemp){
-
-  geometry_container[ZONE_0][INST_0][MESH_0]->SetCustomBoundaryTemperature(iMarker, iVertex, val_WallTemp);
+    
+    geometry_container[ZONE_0][INST_0][MESH_0]->SetCustomBoundaryTemperature(iMarker, iVertex, val_WallTemp);
 }
 
 vector<passivedouble> CDriver::GetVertexHeatFluxes(unsigned short iMarker, unsigned long iVertex) const {
-
-  unsigned long iPoint;
-  unsigned short iDim;
-  su2double Prandtl_Lam  = config_container[ZONE_0]->GetPrandtl_Lam();
-  su2double Gas_Constant = config_container[ZONE_0]->GetGas_ConstantND();
-  su2double Gamma = config_container[ZONE_0]->GetGamma();
-  su2double Gamma_Minus_One = Gamma - 1.0;
-  su2double Cp = (Gamma / Gamma_Minus_One) * Gas_Constant;
-  su2double laminar_viscosity, thermal_conductivity;
-  vector<su2double> GradT (3,0.0);
-  vector<su2double> HeatFlux (3,0.0);
-  vector<passivedouble> HeatFluxPassive (3,0.0);
-
-  bool compressible = (config_container[ZONE_0]->GetKind_Regime() == ENUM_REGIME::COMPRESSIBLE);
-
-  iPoint = geometry_container[ZONE_0][INST_0][MESH_0]->vertex[iMarker][iVertex]->GetNode();
-
-  if(compressible){
-    laminar_viscosity    = solver_container[ZONE_0][INST_0][MESH_0][FLOW_SOL]->GetNodes()->GetLaminarViscosity(iPoint);
-    thermal_conductivity = Cp * (laminar_viscosity/Prandtl_Lam);
-    for(iDim=0; iDim < nDim; iDim++){
-      GradT[iDim] = solver_container[ZONE_0][INST_0][MESH_0][FLOW_SOL]->GetNodes()->GetGradient_Primitive(iPoint, 0, iDim);
-      HeatFlux[iDim] = -thermal_conductivity*GradT[iDim];
+    
+    unsigned long iPoint;
+    unsigned short iDim;
+    su2double Prandtl_Lam  = config_container[ZONE_0]->GetPrandtl_Lam();
+    su2double Gas_Constant = config_container[ZONE_0]->GetGas_ConstantND();
+    su2double Gamma = config_container[ZONE_0]->GetGamma();
+    su2double Gamma_Minus_One = Gamma - 1.0;
+    su2double Cp = (Gamma / Gamma_Minus_One) * Gas_Constant;
+    su2double laminar_viscosity, thermal_conductivity;
+    vector<su2double> GradT (3,0.0);
+    vector<su2double> HeatFlux (3,0.0);
+    vector<passivedouble> HeatFluxPassive (3,0.0);
+    
+    bool compressible = (config_container[ZONE_0]->GetKind_Regime() == ENUM_REGIME::COMPRESSIBLE);
+    
+    iPoint = geometry_container[ZONE_0][INST_0][MESH_0]->vertex[iMarker][iVertex]->GetNode();
+    
+    if(compressible){
+        laminar_viscosity    = solver_container[ZONE_0][INST_0][MESH_0][FLOW_SOL]->GetNodes()->GetLaminarViscosity(iPoint);
```

Comment:
```suggestion
    if(compressible){
        //TODO: This will not work for NEMO or INC(?) cases.
        laminar_viscosity    = solver_container[ZONE_0][INST_0][MESH_0][FLOW_SOL]->GetNodes()->GetLaminarViscosity(iPoint);
```

Reply:
Or can this grab the values in CFVMSolverBase.inl?

Reply:
I am not sure... I haven't touched this code. We added some new functions related to laminar viscosity on the feature_pysu2_fsi_adjoint branch, so I will double check for PR #1750.